### ISSUE TITLE
test_serde_custom_tikv_config is not working as intended when there are skipped serializations

### ISSUE THREAD

**Issue by breeswish** [pingcap/tikv#2971]
In test_serde_custom_tikv_config, we serialize the config and deserialize again to check whether nor not they are equal, which is fine in most cases but will cause problems when there are serialization-skipped fields, for example, the deprecated `end_point_concurrency`. To workaround it, we have to set `end_point_concurrency` to the default value, but in this case we will miss the test of deserialization config different to default one.

**Comment by overvenus** [comment]
Sounds like this one, does `#[serde(skip_serializing)]` help?
https://github.com/pingcap/tikv/blob/133d34e3c0e6ed405c731d1cbf7ffea750aed369/src/raftstore/store/config.rs#L113

**Comment by breeswish** [comment]
@overvenus No, in fact `skip_serializing` is the root cause of this issue.

For example, suppose we have a struct contains field `foo` and `bar`, in which `foo` is `skip_serializing`: `{ foo: "a", bar: "b" }` and the default value is `{ foo: "default_a", bar: "default_b" }`.

When we serialize `{ foo: "a", bar: "b" }` and deserialize again, we will get `{ foo: "default_a", bar: "b" }` thus the test is broken.
